finesse/hardware/temperature/dp9800.py

```diff
@@ -0,0 +1,238 @@
+"""This module provides an interface to DP9800 temperature readers."""
+import logging
+from datetime import datetime
+from decimal import Decimal
+
+from pubsub import pub
+from serial import Serial, SerialException
+
+from .temperature_monitor_base import TemperatureMonitorBase
+
+
+def check_data(data: bytes) -> None:
+    """Perform message integrity checks.
+
+    Check characters that we know should be constant. The message should
+    start with STX, end with NUL and contain an ETX. The exact position
+    of ETX is uncertain since it depends on the width of the BCC.
+
+    Args:
+        data: the message to check
+
+    Raises:
+        DP9800Error: Malformed message received from device
+    """
+    header_width = 2
+    value_width = 8
+    num_values = 9
```

Comment:
As I said before, I don't think we should be hardcoding the number of channels in various places. Just check that the header + terminator are there. If the device ever *did* decide to give us fewer than 9 channels' worth of data (it won't) then we can just pass however many values we get along to the `pubsub` listeners, which is fine.